packages/utils/assert-never/src/assertNever.ts

```diff
@@ -0,0 +1,3 @@
+export function assertNever(x: never): never {
```

Comment:
It looks like this does two things. the x: never clause will cause a build-time break, whereas the throw statement will cause a runtime exception. Maybe consider adding an additional routine that is expectNever which causes the build-time error but allows for execution to continue. The scenario here would be places where things end up being forwarded (like a middle component in a chain where the above and below components both have more options. Just a thought.

Reply:
Yeah there's a package out there that does exactly what you're talking about (https://github.com/aikoven/assert-never). I've never run into a scenario where we've run into the throwing code, actually, so I was wondering what the use case was for having the option to bypass the throw. I can just pull that package in instead so we have that option when we need it